lib/bamboo/test.ex

```diff
@@ -92,13 +130,13 @@ defmodule Bamboo.Test do
   def assert_delivered_email(%Bamboo.Email{} = email) do
     import ExUnit.Assertions
     email = Bamboo.Mailer.normalize_addresses(email)
-    assert_received {:delivered_email, ^email}
+    assert_receive {:delivered_email, ^email}, @timeout
```

Comment:
If you send from another process we need to wait a bit so that we don't get false failures. I could probably extract the timeout as an option at some point, but I think this is ok for now